scripts/AddSynonyms.js

```diff
@@ -0,0 +1,95 @@
+const fs = require('fs');
+const algolia = require('algoliasearch');
+const HttpsAgent = require('agentkeepalive').HttpsAgent;
+const keepaliveAgent = new HttpsAgent({
+  maxSockets: 1,
+  maxKeepAliveRequests: 0, // no limit on max requests per keepalive socket
+  maxKeepAliveTime: 30000, // keepalive for 30 seconds
+});
+const Base = require('./Base.js');
+
+class AddSynonymsScript extends Base {
+  constructor() {
+    super();
+    // Bind class methods
+    this.getSource = this.getSource.bind(this);
+    this.parseBatchSynonymsOptions = this.parseBatchSynonymsOptions.bind(this);
+    this.convertCsvToJson = this.convertCsvToJson.bind(this);
+    this.start = this.start.bind(this);
+    // Define validation constants
+    this.message =
+      '\nExample: $ algolia addsynonyms -a algoliaappid -k algoliaapikey -n algoliaindexname -s sourcefilepath -p batchRulesParams\n\n';
+    this.params = [
+      'algoliaappid',
+      'algoliaapikey',
+      'algoliaindexname',
+      'sourcefilepath',
+    ];
+  }
+
+  getSource(path) {
+    const filepath = this.normalizePath(path);
+    if (!fs.lstatSync(filepath).isFile())
+      throw new Error('Source filepath must target valid synonyms file.');
+    return filepath;
+  }
+
+  parseBatchSynonymsOptions(params) {
+    try {
+      const options = {
+        forwardToReplicas: false,
+        clearExistingSynonyms: false,
+      };
+      if (params === null) return options;
+      else return JSON.parse(params);
+    } catch (e) {
+      throw e;
+    }
+  }
+
+  convertCsvToJson(synonymFile, filepath) {
```

Comment:
I would definitely prefer (b).  About the csvtojson module: I might not fully understand how to use it, but from what I understand it converts column titles into object keys and then stores values appropriately in an array of objects.  That would not work well since synonym csvs have no headers and are just lists of n-way synonyms. They also have to be converted to a particular format to be recognized by our API (include "type":"synonym") which is why I included that short function. I would love to understand what you think some of the edge cases are so that I could address them in the function. Let me know!

Reply:
You make great points. I was thinking that since the `csvtojson` module has [various options](https://github.com/Keyang/node-csvtojson#parameters) that could potentially help accommodate formatting `.csv` files into the desired JSON structure, it might be a preferable way to go, but considering [the limitations you pointed out around passing in synonyms as csv](https://www.algolia.com/doc/guides/managing-results/optimize-search-results/adding-synonyms/how-to/managing-synonyms-from-the-dashboard/#csv) it certainly might be easier and make more sense to write a simple parsing function as you've done.

What do you think about using your existing function, then also adding some context (possibly even a link to [Algolia docs](https://www.algolia.com/doc/guides/managing-results/optimize-search-results/adding-synonyms/how-to/managing-synonyms-from-the-dashboard/#csv)) in the [readme somewhere](https://github.com/algolia/algolia-cli/tree/112997bc35913230fe5b91cf36db3a45fa5cc20a#10-add-synonyms--addsynonyms) that describes the rules/limitations around importing a synonyms csv?